reconcile/utils/saasherder/saasherder.py

```diff
@@ -1614,25 +1614,42 @@ def get_container_images_diff_saas_file(
         for rt in saas_file.resource_templates:
             for target in rt.targets:
                 try:
-                    if not target.image:
+                    if not (target.image or target.images):
                         continue
                     commit_sha = self._get_commit_sha(
                         url=rt.url,
                         ref=target.ref,
                         github=github,
                     )
                     desired_image_tag = commit_sha[: rt.hash_length or self.hash_length]
-                    # don't trigger if image doesn't exist
-                    image_registry = f"{target.image.org.instance.url}/{target.image.org.name}/{target.image.name}"
-                    image_uri = f"{image_registry}:{desired_image_tag}"
-                    image_auth = self._initiate_image_auth(saas_file)
-                    error_prefix = f"[{saas_file.name}/{rt.name}] {target.ref}:"
-                    image = self._get_image(
-                        image_uri, saas_file.image_patterns, image_auth, error_prefix
+
+                    image_registries = []
+                    image_uris = []
+                    all_exist = True
+                    wait_for_images = (
+                        [target.image] if target.image else target.images or []
```

Comment:
Could you please throw an error somewhere if both `image` and `images` are set?

Reply:
@chassing I reckon a warning might be better? Something like "Hey, we're using `image` and not `images`"  wdyt?

Reply:
Initially just wanted to have `image` to silently take precedence over `images` if both exist. But you are right: lets be more transparent about this and raise that something is odd.

Note, that the goal is to have `image` removed completely once everything is migrated to `images`.

Reply:
The migration will be swift - will try with tenant first who needs this feature. Once it works for them, I will move all other targets in app-interface via migration script.

Reply:
Sounds good. Please add me with Glitchtip to the earlier adopters list.

Reply:
can we also use `oneOf` in schema to ensure no `image` and `images` set at the same time?

Reply:
For now its easier for me to just raise an exception here, as I dont see a smooth way of adding `oneOf` without duplicating the whole target object in the schema :thinking: 
This is also just a temporary co-existence during the migration time - I will remove `image` and the exception once migration is done.